watcher/local/watcher.go

```diff
@@ -354,14 +364,61 @@ func (ch *ch) handleEventsFromChain(registerer channel.Registerer, progresser ch
 	}
 }
 
+func (ch *ch) handleRegisteredEvent(
+	ctx context.Context,
+	e *channel.RegisteredEvent,
+	registerer channel.Registerer,
+	chRegistry *registry,
+) {
+	parent := ch
+	if ch.isSubChannel() {
+		parent = ch.parent
+	}
+
+	// This lock ensures, when there are one or more sub-channels and
+	// adjudicator event is received for each channel, the events are
+	// processed one after the other.
+	//
+	// Even if older states have been registered for more than one
+	// channel in the same family, the channel tree will be registered
+	// only once.
+	parent.subChsAccess.Lock()
+	defer parent.subChsAccess.Unlock()
+
+	log := log.WithFields(log.Fields{"ID": e.ID(), "Version": e.Version()})
+	log.Debug("Received registered event from chain")
+
+	latestTx := ch.txRetriever.retrieve()
+	log.Debugf("Latest version is (%d)", latestTx.Version)
+
+	higherVersionAvailable := e.Version() < latestTx.Version && e.Version() >= ch.registeredVersion
+	unregisteredMultiLedger := ch.multiLedger && (!ch.registered || ch.registeredVersion < e.Version())
+	if higherVersionAvailable || unregisteredMultiLedger {
+		log.Debugf("Registering latest version (%d)", latestTx.Version)
+		err := registerDispute(ctx, chRegistry, registerer, parent)
+		if err != nil {
+			log.Error("Error registering dispute: ", err)
+			return
+		}
+		log.Debug("Registered successfully")
+		ch.registered = true
```

Comment:
Why is ch.registered only set here?

Reply:
This is the only place where we call `registerDispute`. Where would you expect to set registered elsewhere?